code/modules/metacoin/metacoin.dm

```diff
@@ -33,6 +33,9 @@
 		CRASH("Metacoin amount fetched before value initialized")
 	return metabalance_cached
 
+/client/proc/get_metabalance_async()
+	return metabalance_cached || get_metabalance_db()
+
 /// Gets the user's metabalance from the DB. Blocking.
 /client/proc/get_metabalance_db()
```

Comment:
I think you can put PRIVATE proc here since we don't want this proc to be called by anything but async proc.